DashAI/back/dataloaders/classes/dataloader.py

```diff
@@ -0,0 +1,70 @@
+from abc import abstractmethod
+
+from datasets import ClassLabel
+
+from DashAI.back.config_object import ConfigObject
+
+
+class DataLoader(ConfigObject):
+    """
+    Abstract class with base methods for all data loaders
+    """
+
+    @abstractmethod
+    def load_data(self, dataset_path, url=None):
+        pass
+
+    @abstractmethod
+    def set_task_format(self, task):
+        pass
+
+    def split_dataset(self, dataset, params, class_column):
+        """
+        Split the dataset in train, test and validation data
+        """
+        stratify_column = None
+        if params["stratify"]:
+            stratify_column = class_column
+        test_val = params["test_size"] + params["val_size"]
+        val_size = params["val_size"] / test_val
+        train_split = dataset["train"].train_test_split(
+            test_size=test_val,
+            shuffle=params["shuffle"],
+            seed=params["seed"],
+            stratify_by_column=stratify_column,
+        )
+        test_valid_split = train_split["test"].train_test_split(
+            test_size=val_size,
+            shuffle=params["shuffle"],
+            seed=params["seed"],
+            stratify_by_column=stratify_column,
+        )
+        dataset["train"] = train_split["train"]
+        dataset["test"] = test_valid_split["train"]
+        dataset["validation"] = test_valid_split["test"]
+        return dataset
+
+    def set_classes(self, dataset, class_index):
```

Comment:
If the function is public, then document better and type in if possible what the parameters are.
```suggestion
    def set_classes(self, dataset, class_index: int):
```